mono/mini/method-to-ir.c

```diff
@@ -6620,6 +6620,9 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 			}
 
 			if (check_this) {
+				// MONO_EMIT_NULL_CHECK will only throw an exception for platforms that request explicit-null-check
+				MONO_EMIT_NULL_CHECK(cfg, sp [0]->dreg);
+
 				MonoInst *check;
```

Comment:
Does this actually build on all the platforms? Don't you need the MonoInst *check variable declared before any statements? Otherwise LGTM.